packages/sw-background-sync-queue/src/lib/request-manager.js

```diff
@@ -0,0 +1,76 @@
+import {putResponse} from './response-manager';
+import {getFetchableRequest} from './queue-utils';
+import {tagNamePrefix} from './constants';
+/**
+ * Class to handle all the request related
+ * transformations, replaying, event handling
+ * broadcasting back to controlled pages etc.
+ * @class
+ */
+class RequestManager {
+	/**
+	 * Initializes the request manager
+	 * stores the callbacks object, maintins config and
+	 * attaches event handler
+	 * @param {any} {callbacks, queue}
+	 *
+	 * @memberOf RequestManager
+	 */
+	constructor({callbacks, queue}) {
```

Comment:
You could do:

```
if (callbacks) {
  assert.isInstance({callbacks}, Array);
}
```

You could go even further and check each item in `callbacks` to make sure it's a `function`, but that might be going too far.

The overall goal is to provide guidance during development when folks pass in the wrong parameter types.